UI enhancements

LironRS opened the pull request:
Fixed Issues:
#431 New sidebar "menu" icon still does nothing
#370 Create a toggle and a dots view
#334 Create new CSS markers
#299 Replace the multiple marker icon

Known Issues:
#443 Fix Grey Area on Sidebar Toggle via Google Maps API
#444 Add Accident Icons to Sidebar List 

<!---
@huboard:{"order":435.0,"milestone_order":445,"custom_state":""}
-->

danielhers commented on static/css/markers.css:
```diff
@@ -0,0 +1,120 @@
+.gmnoprint[title*='תאונה'],.gmnoprint[title*='תאונות']{
```
Will have to be adapted when we have translations, I suppose: #438

danielhers commented on static/js/app.js:
```diff
@@ -442,6 +438,21 @@ $(function () {
+            toggleBGDiv.innerHTML = $("#toggle-control").html();
+
+            var toggleDiv = document.createElement('div');
+            toggleDiv.className = "map-button toggle-control pin";
```
Must have a tooltip since it's not at all clear what this does

danielhers commented:
In the "small icons" view the tooltips end with a period, whereas in the "large icons" view they don't. I don't think a period is required in either case.

danielhers commented:
I like the new icons.
You can delete the old icon files if they're not in use anymore.

danielhers commented:
The "my location" icon is still in the old style. Perhaps it can also be changed to fit with the rest?

danielhers commented on static/js/app.js:
```diff
@@ -801,6 +814,12 @@ $(function () {
+            this.markerIconType = !this.markerIconType;
+            this.resetMarkers();
+            this.fetchMarkers();
+            console.log($(".gmnoprint").find());
```
Still necessary?

danielhers commented:
I went through the tour and at some point it showed the old "multiple markers" icon. Need to see why that happens.
(The tour needs fixing to work with the new sidebar in general but that's a different issue. Do we have an issue for it, by the way?)

danielhers commented:
The infowindow pops from the bottom of the marker icon, partially concealing it:
![selection_087](https://cloud.githubusercontent.com/assets/263363/10013715/8531ce52-611c-11e5-9b33-e3faeea58b13.png)
Could it be slightly shifted upwards?

danielhers commented:
Creating new discussions doesn't work! And I don't see any error message.

danielhers commented:
See all comments above. For slight UI improvements you can open an issue and handle later, but the critical ones need to be fixed before I merge.

LironRS commented:
@danielhers, thanks for the code review! I've been away for a while, so some issues are due to integration issues with the latest dev version. I'll fix these issues asap and keep you posted.

danielhers commented:
Great, thanks.
Note that you'll also have to merge the conflicts with your other pull request, that I merged...

LironRS commented:
**1: Will have to be adapted when we have translations, I suppose: #438**
Correct, although the current solution with the CSS markers uses this selector so we have to think of a unique identifier suitable to all languages.

**2: Must have a tooltip since it's not at all clear what this does**
Added a title attribute, if we want to add a proper tooltip then the location needs to change.

**3: In the "small icons" view the tooltips end with a period, whereas in the "large icons" view they don't. I don't think a period is required in either case.**
Unfortunately this is by design, meaning that this is my way of enabling a correct selector for CSS properties (same as 1). Since the title attribute is the only attribute passed to client after Google Maps rendering, there was no other option except identifying the icons by their title. If you have any other suggestions, I'd love to hear it. I suggest sitting down and defining a proper way to identify each icon (single/multiple, accident type, severity, location accuracy) in order to have a cross-language identifier.

**4: I like the new icons. You can delete the old icon files if they're not in use anymore.**
Done.

**5: The "my location" icon is still in the old style. Perhaps it can also be changed to fit with the rest?**
Fixed.

**6: Still necessary?**
Removed console.log line

**7: I went through the tour and at some point it showed the old "multiple markers" icon. Need to see why that happens.**
We need to re-do the tour, add the newly added options, on my end it doesn't even show most of the steps in the correct position.

**8: The infowindow pops from the bottom of the marker icon, partially concealing it:**
This is on purpose, because we want the bottom of the pin to be exactly where the happened occurred, same as the dot markers, therefore if we want to change it we should move the infowindow up, and not the marker down.

**9: Creating new discussions doesn't work! And I don't see any error message.**
Works now :) There's a minor bug that the creating new discussion makes a dot marker and not a pin one, we need to refresh markers after submission.

danielhers commented on process.py:
```diff
@@ -409,7 +409,11 @@ def main():
                                                  title='Please select a directory')
         else:
             dir_name = raw_input('Please provide the directory path: ')
+<<<<<<< HEAD
```
I think you still have some merge leftovers.

danielhers commented on static/css/markers.css:
```diff
@@ -0,0 +1,172 @@
+    transform: rotate(45deg);
+}
+
+.gmnoprint[title$='.']{
```
I see what you mean about the title ending with a dot being the selector. There has to be a better way but this will do for now.

LironRS commented:
@danielhers I agree and would love to hear suggestions.. I tried three other options to make custom marker, none of them is as versatile as this. Anyway, removed the merge leftovers.
